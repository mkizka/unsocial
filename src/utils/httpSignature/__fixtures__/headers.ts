// relayActivity.spec.tsからコピー
export const expectedHeader = {
  accept: "application/activity+json",
  "content-type": "application/json",
  date: "Wed, 01 Jan 2020 00:00:00 GMT",
  digest: "SHA-256=GJpn7Z0rhlxi7/yIwjEeQln6ix6FqsnOBpzQ156l7mM=",
  host: "remote.example.com",
  signature: `keyId="https://myhost.example.com/users/dummy_userId/activity#main-key",algorithm="rsa-sha256",headers="(request-target) host date digest",signature="E4xXrCrDwMfcdyVopZCV/xAOCtNYhCpJvbcC9pM+Rv+hXpgtLNsIce0b0sv0t268/IjyR+ZP0gFsiY4MtzYbjO9Q3GXw2g+dF9IMDFrApmtfC9nrSWCqmd+qUQmmUfxC/lY+UyyzAL+kSH6alEQY5VfX64hm0FFgBqWMaSJllTAfUA50/pxXXo/tJog6As8h3Mljsqm3cyCz0hi1Ctb5cdpVGmKherwGGgKfHRdR/fft3jbZ0t2z2lEDleARo8MMnWLLXKITDTsQUZztSxdXA2RnXSoyRdDzv1iLswSgF+WgsJnw2qdPHsDDeHBpmSD7Fbv6LAZF+6wzzJlctg=="`,
  "user-agent": "Unsocial/1.2.3 (myhost.example.com)",
};

export const noKeyIdHeader = {
  accept: "application/activity+json",
  "content-type": "application/json",
  date: "Wed, 01 Jan 2020 00:00:00 GMT",
  digest: "SHA-256=GJpn7Z0rhlxi7/yIwjEeQln6ix6FqsnOBpzQ156l7mM=",
  host: "remote.example.com",
  signature: `algorithm="rsa-sha256",headers="(request-target) host date digest",signature="E4xXrCrDwMfcdyVopZCV/xAOCtNYhCpJvbcC9pM+Rv+hXpgtLNsIce0b0sv0t268/IjyR+ZP0gFsiY4MtzYbjO9Q3GXw2g+dF9IMDFrApmtfC9nrSWCqmd+qUQmmUfxC/lY+UyyzAL+kSH6alEQY5VfX64hm0FFgBqWMaSJllTAfUA50/pxXXo/tJog6As8h3Mljsqm3cyCz0hi1Ctb5cdpVGmKherwGGgKfHRdR/fft3jbZ0t2z2lEDleARo8MMnWLLXKITDTsQUZztSxdXA2RnXSoyRdDzv1iLswSgF+WgsJnw2qdPHsDDeHBpmSD7Fbv6LAZF+6wzzJlctg=="`,
  "user-agent": "Unsocial/1.2.3 (myhost.example.com)",
};

export const noAlgorithmHeader = {
  accept: "application/activity+json",
  "content-type": "application/json",
  date: "Wed, 01 Jan 2020 00:00:00 GMT",
  digest: "SHA-256=GJpn7Z0rhlxi7/yIwjEeQln6ix6FqsnOBpzQ156l7mM=",
  host: "remote.example.com",
  signature: `keyId="https://myhost.example.com/users/dummy_userId/activity#main-key",headers="(request-target) host date digest",signature="E4xXrCrDwMfcdyVopZCV/xAOCtNYhCpJvbcC9pM+Rv+hXpgtLNsIce0b0sv0t268/IjyR+ZP0gFsiY4MtzYbjO9Q3GXw2g+dF9IMDFrApmtfC9nrSWCqmd+qUQmmUfxC/lY+UyyzAL+kSH6alEQY5VfX64hm0FFgBqWMaSJllTAfUA50/pxXXo/tJog6As8h3Mljsqm3cyCz0hi1Ctb5cdpVGmKherwGGgKfHRdR/fft3jbZ0t2z2lEDleARo8MMnWLLXKITDTsQUZztSxdXA2RnXSoyRdDzv1iLswSgF+WgsJnw2qdPHsDDeHBpmSD7Fbv6LAZF+6wzzJlctg=="`,
  "user-agent": "Unsocial/1.2.3 (myhost.example.com)",
};

export const noHeadersHeader = {
  accept: "application/activity+json",
  "content-type": "application/json",
  date: "Wed, 01 Jan 2020 00:00:00 GMT",
  digest: "SHA-256=GJpn7Z0rhlxi7/yIwjEeQln6ix6FqsnOBpzQ156l7mM=",
  host: "remote.example.com",
  signature: `keyId="https://myhost.example.com/users/dummy_userId/activity#main-key",algorithm="rsa-sha256",signature="E4xXrCrDwMfcdyVopZCV/xAOCtNYhCpJvbcC9pM+Rv+hXpgtLNsIce0b0sv0t268/IjyR+ZP0gFsiY4MtzYbjO9Q3GXw2g+dF9IMDFrApmtfC9nrSWCqmd+qUQmmUfxC/lY+UyyzAL+kSH6alEQY5VfX64hm0FFgBqWMaSJllTAfUA50/pxXXo/tJog6As8h3Mljsqm3cyCz0hi1Ctb5cdpVGmKherwGGgKfHRdR/fft3jbZ0t2z2lEDleARo8MMnWLLXKITDTsQUZztSxdXA2RnXSoyRdDzv1iLswSgF+WgsJnw2qdPHsDDeHBpmSD7Fbv6LAZF+6wzzJlctg=="`,
  "user-agent": "Unsocial/1.2.3 (myhost.example.com)",
};

export const noSignatureHeader = {
  accept: "application/activity+json",
  "content-type": "application/json",
  date: "Wed, 01 Jan 2020 00:00:00 GMT",
  digest: "SHA-256=GJpn7Z0rhlxi7/yIwjEeQln6ix6FqsnOBpzQ156l7mM=",
  host: "remote.example.com",
  "user-agent": "Unsocial/1.2.3 (myhost.example.com)",
};

export const unSupportedDigestHeader = {
  accept: "application/activity+json",
  "content-type": "application/json",
  date: "Wed, 01 Jan 2020 00:00:00 GMT",
  digest: "SHA-512=GJpn7Z0rhlxi7/yIwjEeQln6ix6FqsnOBpzQ156l7mM=",
  host: "remote.example.com",
  signature: `keyId="https://myhost.example.com/users/dummy_userId/activity#main-key",algorithm="rsa-sha256",headers="(request-target) host date digest",signature="E4xXrCrDwMfcdyVopZCV/xAOCtNYhCpJvbcC9pM+Rv+hXpgtLNsIce0b0sv0t268/IjyR+ZP0gFsiY4MtzYbjO9Q3GXw2g+dF9IMDFrApmtfC9nrSWCqmd+qUQmmUfxC/lY+UyyzAL+kSH6alEQY5VfX64hm0FFgBqWMaSJllTAfUA50/pxXXo/tJog6As8h3Mljsqm3cyCz0hi1Ctb5cdpVGmKherwGGgKfHRdR/fft3jbZ0t2z2lEDleARo8MMnWLLXKITDTsQUZztSxdXA2RnXSoyRdDzv1iLswSgF+WgsJnw2qdPHsDDeHBpmSD7Fbv6LAZF+6wzzJlctg=="`,
  "user-agent": "Unsocial/1.2.3 (myhost.example.com)",
};

export const invalidDateHeader = {
  accept: "application/activity+json",
  "content-type": "application/json",
  date: "Wed, 01 Jan 2020 00:00:01 GMT",
  digest: "SHA-256=GJpn7Z0rhlxi7/yIwjEeQln6ix6FqsnOBpzQ156l7mM=",
  host: "remote.example.com",
  signature: `keyId="https://myhost.example.com/users/dummy_userId/activity#main-key",algorithm="rsa-sha256",headers="(request-target) host date digest",signature="E4xXrCrDwMfcdyVopZCV/xAOCtNYhCpJvbcC9pM+Rv+hXpgtLNsIce0b0sv0t268/IjyR+ZP0gFsiY4MtzYbjO9Q3GXw2g+dF9IMDFrApmtfC9nrSWCqmd+qUQmmUfxC/lY+UyyzAL+kSH6alEQY5VfX64hm0FFgBqWMaSJllTAfUA50/pxXXo/tJog6As8h3Mljsqm3cyCz0hi1Ctb5cdpVGmKherwGGgKfHRdR/fft3jbZ0t2z2lEDleARo8MMnWLLXKITDTsQUZztSxdXA2RnXSoyRdDzv1iLswSgF+WgsJnw2qdPHsDDeHBpmSD7Fbv6LAZF+6wzzJlctg=="`,
  "user-agent": "Unsocial/1.2.3 (myhost.example.com)",
};

export const invalidDigestHeader = {
  accept: "application/activity+json",
  "content-type": "application/json",
  date: "Wed, 01 Jan 2020 00:00:00 GMT",
  digest: "SHA-256=GJpn7Z0rhlxi7/yIwjEeQln6ix6FqsnOBpzQ156l7mM=__invalid__",
  host: "remote.example.com",
  signature: `keyId="https://myhost.example.com/users/dummy_userId/activity#main-key",algorithm="rsa-sha256",headers="(request-target) host date digest",signature="E4xXrCrDwMfcdyVopZCV/xAOCtNYhCpJvbcC9pM+Rv+hXpgtLNsIce0b0sv0t268/IjyR+ZP0gFsiY4MtzYbjO9Q3GXw2g+dF9IMDFrApmtfC9nrSWCqmd+qUQmmUfxC/lY+UyyzAL+kSH6alEQY5VfX64hm0FFgBqWMaSJllTAfUA50/pxXXo/tJog6As8h3Mljsqm3cyCz0hi1Ctb5cdpVGmKherwGGgKfHRdR/fft3jbZ0t2z2lEDleARo8MMnWLLXKITDTsQUZztSxdXA2RnXSoyRdDzv1iLswSgF+WgsJnw2qdPHsDDeHBpmSD7Fbv6LAZF+6wzzJlctg=="`,
  "user-agent": "Unsocial/1.2.3 (myhost.example.com)",
};

export const invalidHostHeader = {
  accept: "application/activity+json",
  "content-type": "application/json",
  date: "Wed, 01 Jan 2020 00:00:00 GMT",
  digest: "SHA-256=GJpn7Z0rhlxi7/yIwjEeQln6ix6FqsnOBpzQ156l7mM=",
  host: "invalid.example.com",
  signature: `keyId="https://myhost.example.com/users/dummy_userId/activity#main-key",algorithm="rsa-sha256",headers="(request-target) host date digest",signature="E4xXrCrDwMfcdyVopZCV/xAOCtNYhCpJvbcC9pM+Rv+hXpgtLNsIce0b0sv0t268/IjyR+ZP0gFsiY4MtzYbjO9Q3GXw2g+dF9IMDFrApmtfC9nrSWCqmd+qUQmmUfxC/lY+UyyzAL+kSH6alEQY5VfX64hm0FFgBqWMaSJllTAfUA50/pxXXo/tJog6As8h3Mljsqm3cyCz0hi1Ctb5cdpVGmKherwGGgKfHRdR/fft3jbZ0t2z2lEDleARo8MMnWLLXKITDTsQUZztSxdXA2RnXSoyRdDzv1iLswSgF+WgsJnw2qdPHsDDeHBpmSD7Fbv6LAZF+6wzzJlctg=="`,
  "user-agent": "Unsocial/1.2.3 (myhost.example.com)",
};

export const invalidSignatureHeader = {
  accept: "application/activity+json",
  "content-type": "application/json",
  date: "Wed, 01 Jan 2020 00:00:00 GMT",
  digest: "SHA-256=GJpn7Z0rhlxi7/yIwjEeQln6ix6FqsnOBpzQ156l7mM=",
  host: "remote.example.com",
  signature: `keyId="https://myhost.example.com/users/dummy_userId/activity#main-key",algorithm="rsa-sha256",headers="(request-target) host date digest",signature="__invalid__"`,
  "user-agent": "Unsocial/1.2.3 (myhost.example.com)",
};

export const unSupportedAlgorithmHeader = {
  accept: "application/activity+json",
  "content-type": "application/json",
  date: "Wed, 01 Jan 2020 00:00:00 GMT",
  digest: "SHA-256=GJpn7Z0rhlxi7/yIwjEeQln6ix6FqsnOBpzQ156l7mM=",
  host: "remote.example.com",
  signature: `keyId="https://myhost.example.com/users/dummy_userId/activity#main-key",algorithm="rsa-sha512",headers="(request-target) host date digest",signature="E4xXrCrDwMfcdyVopZCV/xAOCtNYhCpJvbcC9pM+Rv+hXpgtLNsIce0b0sv0t268/IjyR+ZP0gFsiY4MtzYbjO9Q3GXw2g+dF9IMDFrApmtfC9nrSWCqmd+qUQmmUfxC/lY+UyyzAL+kSH6alEQY5VfX64hm0FFgBqWMaSJllTAfUA50/pxXXo/tJog6As8h3Mljsqm3cyCz0hi1Ctb5cdpVGmKherwGGgKfHRdR/fft3jbZ0t2z2lEDleARo8MMnWLLXKITDTsQUZztSxdXA2RnXSoyRdDzv1iLswSgF+WgsJnw2qdPHsDDeHBpmSD7Fbv6LAZF+6wzzJlctg=="`,
  "user-agent": "Unsocial/1.2.3 (myhost.example.com)",
};
