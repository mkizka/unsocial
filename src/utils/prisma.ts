// Stryker disable all
import { PrismaClient } from "@prisma/client";

import { env } from "./env";

// https://www.prisma.io/docs/guides/other/troubleshooting-orm/help-articles/nextjs-prisma-client-dev-practices#solution
const globalForPrisma = globalThis as unknown as {
  prisma: PrismaClient | undefined;
};

export const prisma =
  globalForPrisma.prisma ??
  new PrismaClient({
    log: ["query"],
  });

if (env.NODE_ENV !== "production") globalForPrisma.prisma = prisma;
