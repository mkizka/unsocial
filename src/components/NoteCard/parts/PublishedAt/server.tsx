import { Client } from "./client";
import { format } from "./utils";

type Props = {
  href: string;
  createdAt: Date;
};

export function PublishedAt(props: Props) {
  // react-hydration-error が出るので、初期値のみサーバーコンポーネントで計算してから渡す
  return <Client {...props} initialText={format(props.createdAt)} />;
}
