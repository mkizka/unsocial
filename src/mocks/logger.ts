import { mockDeep } from "jest-mock-extended";

import { createLogger } from "@/utils/logger";

const mockedLogger = mockDeep<ReturnType<typeof createLogger>>();

jest.mock("@/utils/logger");
const mockedCreateLogger = jest.mocked(createLogger);
mockedCreateLogger.mockReturnValue(mockedLogger);

export { mockedLogger };
