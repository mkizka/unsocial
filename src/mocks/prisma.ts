// Stryker disable all
import type { PrismaClient } from "@prisma/client";
import type { DeepMockProxy } from "jest-mock-extended";
import { mockDeep, mockReset } from "jest-mock-extended";

import { prisma } from "@/app/_shared/utils/prisma";

jest.mock("@/app/_shared/utils/prisma", () => ({
  prisma: mockDeep<PrismaClient>(),
}));

beforeEach(() => {
  mockReset(mockedPrisma);
});

// https://www.prisma.io/docs/guides/testing/unit-testing#mocking-the-prisma-client
export const mockedPrisma = prisma as unknown as DeepMockProxy<PrismaClient>;
