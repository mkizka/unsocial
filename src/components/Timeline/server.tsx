import { noteService } from "@/server/service";

import { ClientComponent } from "./client";

export async function Timeline() {
  const notes = await noteService.findManyNoteCards({ count: 10 });
  return <ClientComponent firstLoadedNotes={notes} />;
}
