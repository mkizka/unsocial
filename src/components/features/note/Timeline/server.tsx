import { noteService } from "@/server/service";

import { ClientComponent } from "./client";

type Props = {
  userId?: string;
};

export async function Timeline({ userId }: Props) {
  const notes = await noteService.findManyNoteCards({ userId, count: 10 });
  return <ClientComponent firstLoadedNotes={notes} />;
}
