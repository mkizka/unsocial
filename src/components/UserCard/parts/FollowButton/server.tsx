import { revalidatePath } from "next/cache";

import { prisma } from "@/server/prisma";
import { getServerSession } from "@/utils/getServerSession";

import { action } from "./action";
import { FollowButton as Client } from "./client";

export async function FollowButton({ followeeId }: { followeeId: string }) {
  const session = await getServerSession();
  const sessionUserId = session?.user?.id;
  if (!sessionUserId) {
    return null;
  }
  const follow = await prisma.follow.findFirst({
    where: {
      followeeId,
      followerId: sessionUserId,
    },
  });

  async function handleClick() {
    "use server";
    await action({ followeeId });
    revalidatePath(`/users/${followeeId}`);
  }

  return <Client follow={follow} onClick={handleClick} />;
}
