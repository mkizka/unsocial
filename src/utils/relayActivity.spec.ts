import type { Credential } from "@prisma/client";
import { rest } from "msw";

import { mockedPrisma } from "@/mocks/prisma";
import { server } from "@/mocks/server";

import { mockedKeys } from "./httpSignature/__fixtures__/keys";
import {
  relayActivityToFollowers,
  relayActivityToInboxUrl,
} from "./relayActivity";

jest.useFakeTimers();
jest.setSystemTime(new Date("2020-01-02T00:00:00"));

describe("relayActivity", () => {
  describe("relayActivityToInboxUrl", () => {
    test("正常系", async () => {
      // arrange
      const dummyUserId = "dummy_userId";
      mockedPrisma.credential.findUnique.mockResolvedValueOnce({
        privateKey: mockedKeys.privateKey,
      } as Credential);
      const headerFn = jest.fn();
      const bodyFn = jest.fn();
      server.use(
        rest.post("https://remote.example.com/inbox", async (req, res, ctx) => {
          headerFn(req.headers.all());
          bodyFn(await req.json());
          return res.once(ctx.status(202));
        }),
      );
      // act
      await relayActivityToInboxUrl({
        userId: dummyUserId,
        inboxUrl: new URL("https://remote.example.com/inbox"),
        // @ts-ignore
        activity: { type: "Dummy" },
      });
      // assert
      expect(headerFn.mock.calls[0][0]).toMatchSnapshot();
      expect(bodyFn.mock.calls[0][0]).toEqual({ type: "Dummy" });
    });
  });

  describe("relayActivityToFollowers", () => {
    test("正常系", async () => {
      // arrange
      const dummyUserId = "dummy_userId";
      mockedPrisma.follow.findMany.mockResolvedValue([
        {
          // @ts-ignore
          follower: { inboxUrl: "https://remote1.example.com/users/foo/inbox" },
        },
        // @ts-ignore
        { follower: { inboxUrl: "https://remote2.example.com/inbox" } },
        // @ts-ignore
        { follower: { inboxUrl: "https://remote2.example.com/inbox" } },
        // @ts-ignore
        { follower: { inboxUrl: null } },
      ]);
      mockedPrisma.credential.findUnique.mockResolvedValueOnce({
        privateKey: mockedKeys.privateKey,
      } as Credential);
      const headerFn = jest.fn();
      const bodyFn = jest.fn();
      const inbox1 = rest.post(
        "https://remote1.example.com/users/foo/inbox",
        async (req, res, ctx) => {
          headerFn(req.headers.all());
          bodyFn(await req.json());
          return res.once(ctx.status(202));
        },
      );
      const inbox2 = rest.post(
        "https://remote2.example.com/inbox",
        async (req, res, ctx) => {
          headerFn(req.headers.all());
          bodyFn(await req.json());
          return res.once(ctx.status(202));
        },
      );
      server.use(inbox1, inbox2);
      // act
      await relayActivityToFollowers({
        userId: dummyUserId,
        // @ts-ignore
        activity: { type: "Dummy" },
      });
      // assert
      expect(headerFn.mock.calls[0][0]).toMatchSnapshot();
      expect(headerFn.mock.calls[1][0]).toMatchSnapshot();
      expect(bodyFn.mock.calls[0][0]).toEqual({ type: "Dummy" });
      expect(bodyFn.mock.calls[1][0]).toEqual({ type: "Dummy" });
    });
  });
});
