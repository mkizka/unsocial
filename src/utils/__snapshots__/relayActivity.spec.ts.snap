// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`relayActivity relayActivityToFollowers 正常系 1`] = `
{
  "accept": "application/activity+json",
  "content-type": "application/json",
  "date": "Wed, 01 Jan 2020 15:00:00 GMT",
  "digest": "SHA-256=GJpn7Z0rhlxi7/yIwjEeQln6ix6FqsnOBpzQ156l7mM=",
  "host": "remote1.example.com",
  "signature": "keyId="https://myhost.example.com/users/dummy_userId/activity#main-key",algorithm="rsa-sha256",headers="(request-target) host date digest",signature="HP4z0xdQEgCzB1ayHVyMA/HgXO+SV5PyZtfVhB7Fc2FuFdYR5nQVgn86ZgA12gyqfpvekAyKfEu/KHAjiU0OUoP7XzHQi4R368rH2MSFBLGetMr6ih4gCjpLv5qdjNRZimM13scz9Dasxt4QVa5WoRIYgAwcdpEzBpuGhcewDlcRcT/Lnwd7JHZGPNcVMdnLRn7zsksEr1wwLrkwplSNHVDCmwUOAmpt3SmQbm6d+jtrDTebY5pJCkwKYkWhWEdED80/5ScXKszbPypUiER2gOITjv+Z52b8Z+5JPbDKfsvHfxkubZjxhAS9MaddAdXEpjAoiFywL4dyqVWMiQ=="",
}
`;

exports[`relayActivity relayActivityToFollowers 正常系 2`] = `
{
  "accept": "application/activity+json",
  "content-type": "application/json",
  "date": "Wed, 01 Jan 2020 15:00:00 GMT",
  "digest": "SHA-256=GJpn7Z0rhlxi7/yIwjEeQln6ix6FqsnOBpzQ156l7mM=",
  "host": "remote2.example.com",
  "signature": "keyId="https://myhost.example.com/users/dummy_userId/activity#main-key",algorithm="rsa-sha256",headers="(request-target) host date digest",signature="dUEV0rTeHU2l60P+1oIB9V+59fu2KW7oINiJ8c6JWj0XuTfR77RouOjDa7ydD194FbY6noGEg9NmrbaiYHvAsNq6QWlq4jN9vee1a5U51WrD9/8vduhxxL11SdxxiqRFlnD6J5/hcYWw3+LuOgsdD366eA7qgDFTj43WxelRAPhl7ZTbP6PX08ZauIWhdmVoSWZnn4ENgSg7rpEGZrwOSLYBH9sRqIBbcNhb9w1qlfdjnaDpDLSqyBRV2xzUzj/AWO1Rc7pfGTzKCuFxsYXVKkcx2aVFI71ztN68OjJm6lZmxf6QBBeD+xoT31rVh93u/TgGW24lwxhVSDDU1A=="",
}
`;

exports[`relayActivity relayActivityToInboxUrl 正常系 1`] = `
{
  "accept": "application/activity+json",
  "content-type": "application/json",
  "date": "Wed, 01 Jan 2020 15:00:00 GMT",
  "digest": "SHA-256=GJpn7Z0rhlxi7/yIwjEeQln6ix6FqsnOBpzQ156l7mM=",
  "host": "remote.example.com",
  "signature": "keyId="https://myhost.example.com/users/dummy_userId/activity#main-key",algorithm="rsa-sha256",headers="(request-target) host date digest",signature="MYlsSoVsAnRAH6+d9DNInUw0aeLPON9K297b1Plm4V2F0zeKGv46sZMhFXxvci63WTslCJRjjHPxFDMajizNjN3dBDzVCZlVhY/eamXDTJwyYl2efmuckxtd1WCTlbFOfE4DD5Ud10E30b/DxhnVXL6ZFWdQofR+FrRXgsEmLwHIyFQbXgdUnSmGi0gEYVkrb/JolnBbX/Leu3IhaZ7osAUgM2txJB8eEL0QqeYqRwf1ZD2QPH6EVE5rngb50gLaxzlNqjutKgDdoeYY+4mV11FsH6HzkrDcJuTxRsqjiffSYXKU1QJTC/etiLaGtK71lBsq+ovu1Nl6oczNCQ=="",
}
`;
